### PR TITLE
Make `Drop[]` of  0th not error, DRY "normal" error messages,  and go over `Drop` doc

### LINKED ISSUE #1223: `Drop[]` 0 element  returns an error when it should not
## Description

`Drop[]` of the 0th element returns an error when it should not

## How to Reproduce

```
$ mathics -e 'Drop[{2}, 0]'
Drop::drop: Cannot drop positions 0 through Infinity in {2}.
Drop[{2}, 0]
```
## Expected output

```
{2}
```
## Additional context

Encountered while debugging Combinatorica V0.9.

### PR DESCRIPTION
**Pull request by rocky**
* Make `Drop[]` of  0th not error 
* DRY "normal" error messages
* Go over `Drop` doc
* Move some doctests to pytests when they belong there
* Move some pytests to less generic and more the more specific module that the test tests

Fixes #1223

### CODE REVIEW

File: mathics/builtin/assignments/assignment.py [hunk 1]
```diff
@@ -355,10 +355,6 @@ class UpSet(InfixOperator, _SetOperator):
     >> UpValues[b]
      = {HoldPattern[a[b]] :> 3}
 
-    >> a ^= 3
```
Comment (rocky) [inline mathics/builtin/assignments/assignment.py:358]: This is moved to pytest. It does not belong as a doctest.

File: mathics/builtin/list/rearrange.py [hunk 2]
```diff
@@ -398,7 +396,6 @@ def eval(self, expr, n, evaluation: Evaluation):
 
 class _SetOperation(Builtin):
     messages = {
-        "normal": "Non-atomic expression expected at position `1` in `2`.",
```
Comment (rocky) [inline mathics/builtin/list/rearrange.py:401]: Notice that in the duplication we have variation in the message: Non-atomic versus Nonatomic

File: mathics/builtin/list/rearrange.py [hunk 3]
```diff
@@ -432,7 +429,7 @@ def eval(self, lists, evaluation, options={}):
                 evaluation.message(
                     self.get_name(),
                     "normal",
-                    pos + 1,
+                    Integer(pos + 1),
```
Comment (rocky) [inline mathics/builtin/list/rearrange.py:432]: There are still a number of inconsistent type parameters in these message calls.

File: test/builtin/test_assignment.py [hunk 4]
```diff
@@ -428,14 +428,6 @@ def test_process_assign_other():
             ],
             "Unset Message",
         ),
-        # From assignent
```
Comment (rocky) [inline test/builtin/test_assignment.py:431]: Moved to the test module where this belongs.

File: test/builtin/test_assignment.py [hunk 5]
```diff
@@ -395,17 +395,17 @@ def test_process_assign_other():
             check_evaluation(
                 f"{prefix}{limit} = 2",
                 "2",
-                expected_messages=[
-                    f"Cannot set {limit} to 2; value must be an integer between 20 and {suffix}."
-                ],
+                expected_messages=(
```
Comment (rocky) [inline test/builtin/test_assignment.py:398]: Type mismatches.

**Comment by rocky** [comment]
I find it sad that after this many years, we are still working on working on getting the calls on methods to pass the correct parameters, DRYing error messages, moving pytests out of doctests, and moving pytests from more general module names into the specific modules that the tests occur in. Sigh.

File: mathics/builtin/assignments/assignment.py [hunk 6]
```diff
@@ -355,10 +355,6 @@ class UpSet(InfixOperator, _SetOperator):
...
-     : Nonatomic expression expected.
-     = 3
-
```
Comment (rocky) [inline mathics/builtin/assignments/assignment.py:361]: This is not quite what WMA reports.

Comment (mmatera) [reply mathics/builtin/assignments/assignment.py:361]:
```
In[1]:= a^=3                                                                    

UpSet::normal: Nonatomic expression expected at position 1 in a ^= 3.

Out[1]= a ^= 3
```
This is what I get in WMA

Comment (mmatera) [reply mathics/builtin/assignments/assignment.py:361]: OK, I see the connection now...

**Review by mmatera — APPROVED** [review]
LGTM

**Pull request merged**